### PR TITLE
Add form for submitting comments

### PR DESCRIPTION
**Pull request by JosephBushagour**
Comments can be submitted and viewed, are not saved in database however. Image of page below:
index_comment_submission

### CODE REVIEW

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 1]
```diff
@@ -47,4 +44,14 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // get comment from form and add it to ArrayList
```
Comment (KeatonTech) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:50]: nit: You don't need this comment. In general comments should explain "why", not "what". The code is clear enough to show the "what" by itself. Same below

File: portfolio/src/main/webapp/index.html [hunk 2]
```diff
@@ -26,8 +26,16 @@ <h1>Joey Bushagour's Portfolio</h1>
       <p>Click here to get a fun fact about me:</p>
       <button onclick="addRandomFunFact()">Here</button>
       <div id="fun-fact-container"></div>
+      
       <h2>Comments:</h2>
       <div id="comment-container"></div>
+
+      <h3>Leave a Comment:</h3>
+      <form action="/data" method="POST">
+      <p>Comment:</p>
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/index.html:35]: nit: Fix indentation

File: portfolio/src/main/webapp/script.js [hunk 3]
```diff
@@ -50,6 +50,10 @@ function addRandomFunFact() {
    const response = await fetch('/data');
    const comments = await response.json();
 
-   // display comments in comment container
-   document.getElementById('comment-container').innerText = comments.join('\n');
+   // determine what to output
+   const output = comments.length ? comments.join('\n')
```
Comment (KeatonTech) [inline portfolio/src/main/webapp/script.js:54]: nit: Kind of odd formatting for this ternary, maybe the ? section should have its own line

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java (hunk 1 above)
Comment (JosephBushagour) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java]: Appreciate the feedback, that's very helpful to know! I've removed those comments and pushed a fix.

File: portfolio/src/main/webapp/index.html (hunk 2 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/index.html]: Good catch! fix pushed.

File: portfolio/src/main/webapp/script.js (hunk 3 above)
Comment (JosephBushagour) [reply portfolio/src/main/webapp/script.js]: Thanks! That's definitely a better choice. Fix pushed.

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -47,4 +44,11 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     response.setContentType("application/json;");
     response.getWriter().println(json);
   }
+
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String comment = request.getParameter("comment");
+    testComments.add(comment);
+    response.sendRedirect("/index.html");
```
Comment (KeatonTech) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: I'm not sure a redirect is what you want here. Usually endpoints like this return an empty response with a 200 status code.

Comment (JosephBushagour) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: The redirect is required (due to the weird implementation in the walkthrough) to move the user back to the page with the comments. Without it the user is stuck on the /data page.

Comment (KeatonTech) [reply portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:52]: Got it, sounds good then

**Pull request merged**